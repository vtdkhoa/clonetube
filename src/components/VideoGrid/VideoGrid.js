import React, { Fragment } from 'react'
import VideoGridHeader from './VideoGridHeader/VideoGridHeader'
import VideoPreview from '../VideoPreview/VideoPreview'
import { Divider } from 'semantic-ui-react'
import './VideoGrid.scss'

const VideoGrid = props => {
  if (!props.videos || !props.videos.length) {
    return <div/>
  }

  const gridItems = props.videos.map(video => {
    return (
      <VideoPreview video={video} key={video.id}/>
    )
  })
  const divider = props.hideDivider ? null : <Divider/>

  return (
    <Fragment>
      <VideoGridHeader title={props.title}/>
      <div className="video-grid">
        {gridItems}
      </div>
      {divider}
    </Fragment>
  )
}

export default VideoGrid