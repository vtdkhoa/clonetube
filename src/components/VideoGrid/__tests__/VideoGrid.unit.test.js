import React from 'react'
import VideoGrid from '../VideoGrid'
import { shallow } from 'enzyme'

describe('VideoGrid', () => {
  test('renders without props', () => {
    const wrapper = shallow(<VideoGrid/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with title prop', () => {
    const wrapper = shallow(
      <VideoGrid title="Trending"/>
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with divider', () => {
    const wrapper = shallow(
      <VideoGrid hideDivider={true}/>
    )
    expect(wrapper).toMatchSnapshot()
  })
})