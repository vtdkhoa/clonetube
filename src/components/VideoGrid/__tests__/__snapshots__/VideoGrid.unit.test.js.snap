// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders with divider 1`] = `<div />`;

exports[`VideoGrid renders with title prop 1`] = `<div />`;

exports[`VideoGrid renders without props 1`] = `<div />`;
